internal/tiltfile/tiltfile_state.go

```diff
@@ -1068,8 +1056,20 @@ func (s *tiltfileState) imgTargetsForDependencyIDsHelper(ids []model.TargetID, c
 		}
 		claimStatus[id] = claimPending
 
+		registry := s.defaultRegistryHost
+		if s.orchestrator() == model.OrchestratorK8s && !s.privateRegistry.Empty() {
+			// If we've found a private registry in the cluster at run-time,
+			// use that instead of the one in the tiltfile
+			s.logger.Infof("Auto-detected private registry from environment: %s", s.privateRegistry)
+			registry = s.privateRegistry
```

Comment:
hmm, I would expect something specified in the Tiltfile to override something found in the environment--do you have thoughts?

Reply:
Nope, if you're using a local registry (like microk8s), you always want Tilt to use that, regardless of what's in the Tiltfile.

Reply:
👍 cool thanks